### PR TITLE
Custom UIScrollView with cinterop

### LINKED ISSUE #489: Remove viewFactory from UIViewRedwoodLayoutWidgetFactory.
`UIViewRedwoodLayoutWidgetFactory` currently accepts a `viewFactory: RedwoodUIScrollViewFactory` param that creates `UIScrollView` instances. This is a work-around as it's not (easily) possible to extend `UIScrollView` from Kotlin multiplatform code so we delegate the creation to Swift code.

This is a pretty inconvenient work-around and we should figure out a way to remove the `viewFactory` param and create `UIScrollView`s without an external factory.

### PR DESCRIPTION
**Pull request by kpgalligan**
There's probably some renaming or moving to do here, but this PR includes a UIScrollView in Objc with a delegate, to avoid copy/pasting code.

Closes #489. Closes #586.

### CODE REVIEW

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/RedwoodUIScrollViewFactory.kt [hunk 1]
```diff
@@ -15,30 +15,17 @@
...
-  public fun setNeedsLayout()
-  public fun layoutSubviews()
-}
-
 /**
  * A factory that creates [UIScrollView] subclasses that delegate to a [RedwoodUIScrollViewDelegate].
  */
-public interface RedwoodUIScrollViewFactory {
-  public fun create(delegate: RedwoodUIScrollViewDelegate): UIScrollView
+public class RedwoodUIScrollViewFactory {
```
Comment (colinrtwhite) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/RedwoodUIScrollViewFactory.kt]: We can remove this too and allocate the `HostView` directly inside `UIViewFlexContainer`.

File: samples/emoji-search/ios/shared/src/commonMain/kotlin/app/cash/zipline/samples/emojisearch/exposed.kt [hunk 2]
```diff
@@ -38,6 +39,7 @@ fun exposedTypes(
   treehouseUIKitView: TreehouseUIKitView<*>,
   uiViewRedwoodLayoutWidgetFactory: UIViewRedwoodLayoutWidgetFactory,
   uiViewRedwoodTreehouseLazyLayoutWidgetFactory: UIViewRedwoodTreehouseLazyLayoutWidgetFactory<*>,
+  redwoodUIScrollViewFactory: RedwoodUIScrollViewFactory,
```
Comment (colinrtwhite) [inline samples/emoji-search/ios/shared/src/commonMain/kotlin/app/cash/zipline/samples/emojisearch/exposed.kt]: I think we can remove this and inline the `HostView` creation in `UIViewFlexContainer`.

**Review by colinrtwhite — APPROVED** [review]
💯 Thanks for fixing this! The iOS folks are going to be stoked to remove the copy/pasted Swift code.

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/RedwoodUIScrollViewFactory.kt (hunk 1 above)
Comment (kpgalligan) [reply redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/RedwoodUIScrollViewFactory.kt]: Updated

File: samples/emoji-search/ios/shared/src/commonMain/kotlin/app/cash/zipline/samples/emojisearch/exposed.kt (hunk 2 above)
Comment (kpgalligan) [reply samples/emoji-search/ios/shared/src/commonMain/kotlin/app/cash/zipline/samples/emojisearch/exposed.kt]: Updated

**Comment by colinrtwhite** [comment]
@kpgalligan Pushed a fix for the compilation issue [here](https://github.com/cashapp/redwood/commit/a381cd98fc57a5f0bdc4dd119e38a7a35f56373a), but when I run the emoji sample app it doesn't render correctly. It seems like `RedwoodScrollViewDelegateProtocol` produces `intrinsicContentSize` and `sizeThatFits` methods with a different signature that use a custom `app.cash.redwood.layout.uiview.cinterop.CGSize`.

**Comment by kpgalligan** [comment]
I was having an issue running the sample locally anyway, so was flying blind a bit. The app ran, but wasn't getting local data properly. Will try to fix that and adjust.

**Comment by kpgalligan** [comment]
Took a bit to get back to this @colinrtwhite. I don't think you need `app.cash.redwood.layout.uiview.cinterop.CGSize`. I got your branch, and replaced it with just `platform.CoreGraphics.CGSize` and it compiles for me.

I updated my branch with your changes to `EmojiSearch.kt`. Here's a screenshot with the output (and some vertical padding on the Column, because the search box was up by the notch).

Simulator Screen Shot - iPhone 13 - 2022-12-12 at 20 10 14

I just tried `import app.cash.redwood.layout.uiview.cinterop.CGSize` in my branch and it fails to compile.

Will need to dig into more tomorrow I guess.

**Comment by kpgalligan** [comment]
> Pushed a fix for the compilation issue [here](https://github.com/cashapp/redwood/commit/a381cd98fc57a5f0bdc4dd119e38a7a35f56373a), but when I run the emoji sample app it doesn't render correctly. It seems like `RedwoodScrollViewDelegateProtocol` produces `intrinsicContentSize` and `sizeThatFits` methods with a different signature that use a custom `app.cash.redwood.layout.uiview.cinterop.CGSize`.

@colinrtwhite I tried your branch again and its the same result as mine. IDE and CLI build don't resolve `import app.cash.redwood.layout.uiview.cinterop.CGSize`. Not sure why you're seeing that, but I can't repro that in either branch.
Screen Shot 2022-12-15 at 4 37 37 PM

I don't want to just dismiss what you're seeing, though, so if you could, please try a clean build or whatever to see if we can figure that out.

**Pull request merged**